improver_tests/utilities/test_save.py

```diff
@@ -149,6 +149,34 @@ def test_compression_level_out_of_range(self):
         with self.assertRaises(ValueError):
             save_netcdf(self.cube, self.filepath, compression_level=10)
 
+    def test_least_significant_digit_1_no_compression(self):
+        """ Test setting the least significant digit to 1 with no compression """
+        save_netcdf(
+            self.cube, self.filepath, compression_level=0, least_significant_digit=1
+        )
+
+        data = Dataset(self.filepath, mode="r")
+        # pylint: disable=unsubscriptable-object
+        lsd = data.variables["air_temperature"].least_significant_digit
+        self.assertEqual(lsd, 1)
+
+        data_cube = load_cube(self.filepath)
+        self.assertArrayAlmostEqual(data_cube.data, self.cube.data, decimal=1)
```

Comment:
Suggest adding a check here that the bit shaving has actually modified the original data. As is, this test looks like it would pass if the `least_significant_digit` option just stored its setting in the metadata and had no effect on the data.

A basic method would be to check that there is a difference between before/after, or a more detailed method would be to check some statistics eg. max/mean/median/min of the relative difference between before/after.

Reply:
Thanks Tom, I've adjusted the tests to ensure a difference in the stats

Reply:
I've created a PR to PR https://github.com/BelligerG/improver/pull/3 with an attempt at better checking of the bitshaved data.